### ISSUE TITLE
Chrome complaining about "illegal use of labels"

### ISSUE THREAD

**Issue by matrium0** [angular/components#28184]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

When using <mat-form-field> with  <mat-label> Chrome prints out the following warning:
**Incorrect use of <label for=FORM_ELEMENT>**

I assume this is a new check Chrome added in a recent versions to it's dev tools. Though I guess Angular Material should probably comply to those, if at all possible.

### Reproduction

This can be easily reproduced, even on the Angular Material Page, e.g. on https://material.angular.io/components/form-field/overview (though it appears on any page that uses <mat-label> within <mat-form-field>

![image](https://github.com/angular/components/assets/36888285/0d444ad5-98a6-470f-99d8-b2d0e6a31c3a)

### Expected Behavior

No error should be displayed

### Actual Behavior

**Incorrect use of <label for=FORM_ELEMENT>** is printed for each mat-label within a mat-form-field

### Environment

- Angular: 17.0.1
- CDK/Material: 17.0.1
- Browser(s): Chrome 117
- Operating System: Windows

**Comment by crisbeto** [comment]
It looks like it's complaining because the `for` attribute is pointing to `<mat-select>` which isn't a native input.

**Comment by matrium0** [comment]
@crisbeto You are right, I didn't not realize that it's only related to mat-select and not input (which makes sense). I updated the ticket to better reflect this.
